analyzers/src/SonarAnalyzer.Common/Rules/ImplementSerializationMethodsCorrectlyBase.cs

```diff
@@ -62,39 +71,26 @@ protected override void Initialize(SonarAnalysisContext context)
                     var location = GetIdentifierLocation(methodSymbol);
                     if (location != null)
                     {
-                        c.ReportDiagnosticIfNonGenerated(GeneratedCodeRecognizer, Diagnostic.Create(SupportedDiagnostics[0],
-                            location, issues.ToSentence()));
+                        c.ReportDiagnosticIfNonGenerated(GeneratedCodeRecognizer, Diagnostic.Create(SupportedDiagnostics[0], location, issues.ToSentence()));
                     }
                 },
                 SymbolKind.Method);
-        }
-
-        protected abstract GeneratedCodeRecognizer GeneratedCodeRecognizer { get; }
-        protected abstract string MethodReturnTypeShouldBeVoidMessage { get; }
 
-        protected abstract Location GetIdentifierLocation(IMethodSymbol methodSymbol);
-
-        private IEnumerable<string> FindIssues(IMethodSymbol methodSymbol)
+        private List<string> FindIssues(IMethodSymbol methodSymbol)
```

Comment:
the return type can stay `IEnumerable<string>` (it's only being enumerated upon, right?)

being private and a short class, I think we can live with it as it is, as well

Reply:
I had to rewrite the `.Count` check. And that inspired me to improve the condition, simplify the flow and remove 6 unneeded rows :) So it's a valuable update.